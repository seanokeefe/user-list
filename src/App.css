html, body, #root, .App {
  height: 100%;
}

#list {
  height: 100%;
  display: flex;
  flex-direction: column;
}

.header {
  background-color: #9CAAC6;
  flex: 0 0 5vh;
  padding: 1rem 2rem;
}

.header h1 {
  color: #ffffff;
  text-transform: uppercase;
  font-size: 1.5rem;
  font-weight: 100;
}

.header h1 span{
  margin: 0 1rem;
}

.main {
  background-color: #DEE7EF;
  justify-content: center;
  display: flex;
  flex: 1 1 85vh;
  flex-wrap: wrap;
  flex-direction: row;
  padding: 1rem 2rem;
  overflow-y: auto;
}

.footer {
  background-color: #0C2C52;
  color: #9CAAC6;
  flex: 0 0 10vh;
  padding: 1rem 2rem;
}

.footer div {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: 0 1rem;
}

.sort {
  display:flex;
  justify-content: flex-end;
  align-items: baseline;
  border: 1px solid #0C2C52;
  border-radius: 5px;
  color: #dedede;
  padding: 0.5em 0.5rem 0.5rem 1rem;
  font-size: 1rem;
}

.sort span {
  font-size: 0.6rem ;
  text-transform: uppercase;
  color: #0C2C52;
  margin-right: 0.5rem;
  color: #0C2C52;
}

.sort span.sort-icons {
  font-size: 1.5rem;
  cursor: pointer;
  padding: 0.15rem 0.3rem;
}

.sort span.sort-icons--active {
  font-size: 1.5rem;
  border: 1px solid #0C2C52;
  border-radius: 5px;
  padding: 0.15rem 0.3rem;
  box-shadow: inset 2px 2px 5px #0C2C52;
}

.sort button {
  padding: 0;
  border: 0;
  background: none;
  margin: 0 0.5rem;
  outline: none;
}

.sort button span {
  margin: 0;
}

.sort select {
  appearance: none;
  background: none;
  border: none;
  color: #ffffff;
  font-size: 1rem;
  outline: none;
  cursor: pointer;
  text-align: right;
  margin-right: 1rem;
}

.list-item {
  display: flex;
  width: 325px;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: center;
  background-color: #ececec;
  color: #003366;
  margin: 1rem 0.5rem;
  padding: 1rem;
  border-radius: 5px;
  box-shadow: 2px 3px 7px #9CAAC6;
}

.list-item__avatar {
  border-radius: 25%;
}

.list-item__lockup {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
}

.list-item__name {
  flex-grow: 1;
  font-weight: 600;
  font-size: 1.25rem;
  margin: 0.5rem 1rem 0.25rem;
}

.list-item__address {
  margin: 0.25rem 1rem;
  color: #336699;
}

.list-item__balance {
  display: flex;
  width: 100%;
  flex-grow: 1;
  justify-content: center;
  margin: 1rem auto 0.5rem;
  border-top: 2px solid #9CAAC6;
  border-bottom: 2px solid #9CAAC6;
  border-radius: 5px;
  background-color: #DEE7EF;
}

.list-item__balance h4{
  color: #627894;
  margin: 0.75rem auto 0.5rem;
  text-transform: uppercase;
}

.list-item__balance h4 span{
  color: #0C2C52;
}

.list-item__created {
  color: #9CAAC6;
  font-size: 0.7rem;
  margin: 0.25rem 0;
  text-align: right;
  flex-grow: 1;
}
