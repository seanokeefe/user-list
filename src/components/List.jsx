import React from 'react';
import PropTypes from 'prop-types';
import cuid from 'cuid';

const List = ({ page, records, orderBy }) => (
  <div>
    {page &&
      <div>
        <div>Total: {page.total}</div>
        <div>Pages: {page.pages}</div>
        <div>Limit: {page.limit}</div>
        <div>Current: {page.current}</div>
      </div>
    }

    {orderBy && orderBy.key && (
      <div>{`Order by ${orderBy.key}, ${orderBy.dir}`}</div>
    )}

    {records && records.map(({ name }) => (
      <div key={cuid()}>{ name }</div>
    ))}
  </div>

);

List.propTypes = {
  page: PropTypes.shape({
    total: PropTypes.number,
    pages: PropTypes.number,
    limit: PropTypes.number,
    current: PropTypes.number,
    previous: PropTypes.number,
    next: PropTypes.number,
    showing: PropTypes.shape({
      start: PropTypes.number,
      end: PropTypes.number,
    }),
  }),
  records: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    avatar: PropTypes.string,
    address: PropTypes.shape({
      street: PropTypes.string,
      city: PropTypes.string,
      zip: PropTypes.sting,
      country: PropTypes.string,
    }),
  })),
  orderBy: PropTypes.shape({
    key: PropTypes.string,
    dir: PropTypes.string,
  }),
};

export default List;
