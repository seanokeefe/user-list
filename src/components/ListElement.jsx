import React from 'react';
import PropTypes from 'prop-types';

const ListElement = ({ name, avatar, address, balance, created }) => {
  const { street, city, zip, country } = address;

  return (
    <div className="list-item">
      <img className="list-item__avatar" src={avatar} alt={`avatar of ${name}`} />
      <div className="list-item__lockup">
        <div className="list-item__name">
          {name}
        </div>
        <div className="list-item__address">
          {street} <br />
          {`${city}, ${country} ${zip}`} <br />
        </div>
      </div>
      <div className="list-item__balance">
        <h4>Balance: <span>{balance}</span></h4>
      </div>
      <div className="list-item__created">
        record created: {created}
      </div>
    </div>
  );
};

ListElement.propTypes = {
  name: PropTypes.string,
  avatar: PropTypes.string,
  address: PropTypes.shape({
    street: PropTypes.string,
    city: PropTypes.string,
    zip: PropTypes.sting,
    country: PropTypes.string,
  }),
  balance: PropTypes.number,
  created: PropTypes.string,
};

export default ListElement;
