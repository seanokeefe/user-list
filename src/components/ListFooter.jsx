import React from 'react';
import PropTypes from 'prop-types';

import PageControl from './PageControl';
import RecordsControl from './RecordsControl';

const ListFooter = ({ total, pages, limit, currentPage, handleLimitChange, handlePageChange }) => (
  <div className="footer">
    <PageControl
      limit={limit}
      currentPage={currentPage}
      pages={pages}
      handlePageChange={handlePageChange}
    />
    <RecordsControl limit={limit} total={total} handleLimitChange={handleLimitChange} />
  </div>
);

ListFooter.propTypes = {
  total: PropTypes.number,
  pages: PropTypes.number,
  limit: PropTypes.number,
  currentPage: PropTypes.number,
  handleLimitChange: PropTypes.func,
  handlePageChange: PropTypes.func,
};

export default ListFooter;
