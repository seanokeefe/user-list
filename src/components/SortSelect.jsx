import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

const SortSelect = ({ sortedBy, dir, handleSortChange, handleDirChange }) => (
  <div className="sort">
    <span>Sort By:</span>
    <select name="orderKey" value={sortedBy} onChange={handleSortChange}>
      <option value="name" >name</option>
      <option value="balance" > balance</option>
      <option value="city" >city</option>
      <option value="country" >country</option>
    </select>
    <div>
      <button name="orderDirection" value="asc" onClick={handleDirChange}><FontAwesome className={`sort-icons${dir === 'asc' ? '--active' : ''}`} name="caret-square-o-up" /></button>
      <button name="orderDirection" value="desc" onClick={handleDirChange}><FontAwesome className={`sort-icons${dir === 'desc' ? '--active' : ''}`} name="caret-square-o-down" /></button>
    </div>
  </div>

);

SortSelect.propTypes = {
  sortedBy: PropTypes.string,
  dir: PropTypes.string,
  handleSortChange: PropTypes.func,
  handleDirChange: PropTypes.func,
};

export default SortSelect;
