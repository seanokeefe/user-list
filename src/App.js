import React from 'react';
import ListContainer from './containers/ListContainer';
import './App.css';

const App = () => (
  <div className="App">
    <ListContainer />
  </div>
);

export default App;
