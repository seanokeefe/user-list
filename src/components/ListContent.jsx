import React from 'react';
import PropTypes from 'prop-types';
import cuid from 'cuid';

import ListElement from './ListElement';

const ListContent = ({ records }) => (
  <div className="main">
    {records && records.map(({ name, email, avatar, address, balance, created }) => (
      <ListElement
        key={cuid()}
        name={name}
        email={email}
        avatar={avatar}
        address={address}
        balance={balance}
        created={created}
      />
    ))}
  </div>
);

ListContent.propTypes = {
  records: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    avatar: PropTypes.string,
    address: PropTypes.shape({
      street: PropTypes.string,
      city: PropTypes.string,
      zip: PropTypes.sting,
      country: PropTypes.string,
    }),
    balance: PropTypes.number,
    created: PropTypes.string,
  })),
};

export default ListContent;
