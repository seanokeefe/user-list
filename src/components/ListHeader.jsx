import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

import SortSelect from './SortSelect';

const ListHeader = ({ sortedBy, dir, handleSortChange, handleDirChange }) => (
  <div className="header">
    <h1>User Records
      <FontAwesome name="folder-open-o" />
    </h1>
    <SortSelect
      sortedBy={sortedBy}
      dir={dir}
      handleSortChange={handleSortChange}
      handleDirChange={handleDirChange}
    />
  </div>
);

ListHeader.propTypes = {
  sortedBy: PropTypes.string,
  dir: PropTypes.string,
  handleSortChange: PropTypes.func,
  handleDirChange: PropTypes.func,
};

export default ListHeader;
