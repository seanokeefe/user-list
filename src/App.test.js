import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from './App';
import ListContainer from './containers/ListContainer';
import List from './components/List';
import ListContent from './components/ListContent';
import ListElement from './components/ListElement';
import ListFooter from './components/ListFooter';
import ListHeader from './components/ListHeader';
import SortSelect from './components/SortSelect';

// eslint-disable-next-line no-undef
test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

// eslint-disable-next-line no-undef
test('checks ListContainer default render', () => {
  const component = renderer.create(
    <ListContainer />,
  );
  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});

// eslint-disable-next-line no-undef
test('checks List default render', () => {
  const component = renderer.create(
    <List />,
  );
  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});

// eslint-disable-next-line no-undef
test('checks ListContent default render', () => {
  const component = renderer.create(
    <ListContent />,
  );
  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});

// eslint-disable-next-line no-undef
test('checks ListElement default render', () => {
  const component = renderer.create(
    <ListElement
      name={''}
      avatar={''}
      address={{ street: '', city: '', zip: '', country: '' }}
      balance={0}
      created={'2082-11-19T09:44:58.655Z'}
    />,
  );
  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});

// eslint-disable-next-line no-undef
test('checks ListFooter default render', () => {
  const component = renderer.create(
    <ListFooter />,
  );
  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});

// eslint-disable-next-line no-undef
test('checks ListHeader default render', () => {
  const component = renderer.create(
    <ListHeader />,
  );
  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});

// eslint-disable-next-line no-undef
test('checks SortSelect default render', () => {
  const component = renderer.create(
    <SortSelect />,
  );
  const tree = component.toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});
