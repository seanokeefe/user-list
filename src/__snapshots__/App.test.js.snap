// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks List default render 1`] = `
<div
  id="list"
>
  <div
    className="header"
  >
    <h1>
      User Records
      <span
        aria-hidden={true}
        className="fa fa-folder-open-o"
      />
    </h1>
    <div
      className="sort"
    >
      <span>
        Sort By:
      </span>
      <select
        name="orderKey"
        onChange={undefined}
        value={undefined}
      >
        <option
          value="name"
        >
          name
        </option>
        <option
          value="balance"
        >
           balance
        </option>
        <option
          value="city"
        >
          city
        </option>
        <option
          value="country"
        >
          country
        </option>
      </select>
      <div>
        <button
          name="orderDirection"
          onClick={undefined}
          value="asc"
        >
          <span
            aria-hidden={true}
            className="fa fa-caret-square-o-up sort-icons"
          />
        </button>
        <button
          name="orderDirection"
          onClick={undefined}
          value="desc"
        >
          <span
            aria-hidden={true}
            className="fa fa-caret-square-o-down sort-icons"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="main"
  />
  <div
    className="footer"
  >
    <div>
      <div>
        Total: 
      </div>
      <div>
        Pages: 
      </div>
      <div>
        Limit: 
      </div>
      <div>
        Current: 
      </div>
    </div>
  </div>
</div>
`;

exports[`checks ListContainer default render 1`] = `
<div
  id="list"
>
  <div
    className="header"
  >
    <h1>
      User Records
      <span
        aria-hidden={true}
        className="fa fa-folder-open-o"
      />
    </h1>
    <div
      className="sort"
    >
      <span>
        Sort By:
      </span>
      <select
        name="orderKey"
        onChange={[Function]}
        value={undefined}
      >
        <option
          value="name"
        >
          name
        </option>
        <option
          value="balance"
        >
           balance
        </option>
        <option
          value="city"
        >
          city
        </option>
        <option
          value="country"
        >
          country
        </option>
      </select>
      <div>
        <button
          name="orderDirection"
          onClick={[Function]}
          value="asc"
        >
          <span
            aria-hidden={true}
            className="fa fa-caret-square-o-up sort-icons"
          />
        </button>
        <button
          name="orderDirection"
          onClick={[Function]}
          value="desc"
        >
          <span
            aria-hidden={true}
            className="fa fa-caret-square-o-down sort-icons"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="main"
  />
  <div
    className="footer"
  >
    <div>
      <div>
        Total: 
      </div>
      <div>
        Pages: 
      </div>
      <div>
        Limit: 
      </div>
      <div>
        Current: 
      </div>
    </div>
  </div>
</div>
`;

exports[`checks ListContent default render 1`] = `
<div
  className="main"
/>
`;

exports[`checks ListElement default render 1`] = `
<div
  className="list-item"
>
  <img
    alt="avatar of "
    className="list-item__avatar"
    src=""
  />
  <div
    className="list-item__lockup"
  >
    <div
      className="list-item__name"
    >
      
    </div>
    <div
      className="list-item__address"
    >
      
       
      <br />
      ,  
       
      <br />
    </div>
  </div>
  <div
    className="list-item__balance"
  >
    <h4>
      Balance: 
      <span>
        0
      </span>
    </h4>
  </div>
  <div
    className="list-item__created"
  >
    record created: 
    
  </div>
</div>
`;

exports[`checks ListFooter default render 1`] = `
<div
  className="footer"
>
  <div>
    <div>
      Total: 
    </div>
    <div>
      Pages: 
    </div>
    <div>
      Limit: 
    </div>
    <div>
      Current: 
    </div>
  </div>
</div>
`;

exports[`checks ListHeader default render 1`] = `
<div
  className="header"
>
  <h1>
    User Records
    <span
      aria-hidden={true}
      className="fa fa-folder-open-o"
    />
  </h1>
  <div
    className="sort"
  >
    <span>
      Sort By:
    </span>
    <select
      name="orderKey"
      onChange={undefined}
      value={undefined}
    >
      <option
        value="name"
      >
        name
      </option>
      <option
        value="balance"
      >
         balance
      </option>
      <option
        value="city"
      >
        city
      </option>
      <option
        value="country"
      >
        country
      </option>
    </select>
    <div>
      <button
        name="orderDirection"
        onClick={undefined}
        value="asc"
      >
        <span
          aria-hidden={true}
          className="fa fa-caret-square-o-up sort-icons"
        />
      </button>
      <button
        name="orderDirection"
        onClick={undefined}
        value="desc"
      >
        <span
          aria-hidden={true}
          className="fa fa-caret-square-o-down sort-icons"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`checks SortSelect default render 1`] = `
<div
  className="sort"
>
  <span>
    Sort By:
  </span>
  <select
    name="orderKey"
    onChange={undefined}
    value={undefined}
  >
    <option
      value="name"
    >
      name
    </option>
    <option
      value="balance"
    >
       balance
    </option>
    <option
      value="city"
    >
      city
    </option>
    <option
      value="country"
    >
      country
    </option>
  </select>
  <div>
    <button
      name="orderDirection"
      onClick={undefined}
      value="asc"
    >
      <span
        aria-hidden={true}
        className="fa fa-caret-square-o-up sort-icons"
      />
    </button>
    <button
      name="orderDirection"
      onClick={undefined}
      value="desc"
    >
      <span
        aria-hidden={true}
        className="fa fa-caret-square-o-down sort-icons"
      />
    </button>
  </div>
</div>
`;
